posthog/models/cohort/util.py

```diff
@@ -449,22 +449,72 @@ def get_dependent_cohorts(
     seen_cohort_ids = set()
     seen_cohort_ids.add(cohort.id)
 
-    queue = [prop.value for prop in cohort.properties.flat if prop.type == "cohort"]
+    queue: List[int] = []
+    for prop in cohort.properties.flat:
+        if prop.type == "cohort" and not isinstance(prop.value, list):
+            queue.append(int(prop.value))
 
     while queue:
         cohort_id = queue.pop()
         try:
-            parsed_cohort_id = str(cohort_id)
-            if parsed_cohort_id in seen_cohorts_cache:
-                cohort = seen_cohorts_cache[parsed_cohort_id]
+            if cohort_id in seen_cohorts_cache:
+                cohort = seen_cohorts_cache[cohort_id]
             else:
                 cohort = Cohort.objects.using(using_database).get(pk=cohort_id)
-                seen_cohorts_cache[parsed_cohort_id] = cohort
+                seen_cohorts_cache[cohort_id] = cohort
             if cohort.id not in seen_cohort_ids:
                 cohorts.append(cohort)
                 seen_cohort_ids.add(cohort.id)
-                queue += [prop.value for prop in cohort.properties.flat if prop.type == "cohort"]
+                for prop in cohort.properties.flat:
+                    if prop.type == "cohort" and not isinstance(prop.value, list):
```

Comment:
curious, why this check? I expect this to never make it past validation, is it not the case?

Reply:
I tried to do this:
`queue += [int(prop.value) for prop in cohort.properties.flat if prop.type == "cohort"]`

The type check gives this error: `Argument 1 to "int" has incompatible type "Union[str, int, List[str]]";`

It seems to think `prop.value` can still be another nested list, even when `type == "cohort"`